File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.js
/**
 * @param {number[]} difficulty
 * @param {number[]} profit
 * @param {number[]} worker
 * @return {number}
 */
var maxProfitAssignment = function(difficulty, profit, worker) {
    worker.sort((a,b) => a - b);
    const jobs = [];

    for (let i = 0; i < profit.length; i++) {
        jobs[i] = i;
    }

    jobs.sort((a,b) =>profit[a] - profit[b]);

    let idx = jobs.length - 1;
    let result = 0;
    for (let i = worker.length - 1; i >= 0; i--) {
        while (worker[i] < difficulty[jobs[idx]]){
            idx--;
            if(idx < 0) {
                return result;
            }
        }
        result += profit[jobs[idx]];
    }
    return result;
};

File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.py
class Solution:
    def maxProfitAssignment(self, difficulty: List[int], profit: List[int], worker: List[int]) -> int:
        worker.sort()
        jobs = [0] * len(profit)
        
        for i in range(len(profit)):
            jobs[i] = i
        
        jobs = sorted(jobs, key=lambda x: profit[x])
        print(jobs)
        
        idx = len(jobs) - 1
        res = 0
        for i in range(len(worker)-1,-1,-1):
            while worker[i] < difficulty[jobs[idx]]:
                idx -= 1
                if idx < 0:
                    return res
            res += profit[jobs[idx]]
        return res